### PR TITLE
Added some tests for useProgressInterval

### PR DESCRIPTION
**Pull request by theianjones**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
     "start": "npm run build -- -w",
     "travis-deploy-once": "npx travis-deploy-once",
     "is-build-es5": "npx es-check es5 './build/**/*.!(esm).js'",
-    "test": "jest"
+    "test": "jest",
+    "test:watch": "jest --watch"
```
Comment (theianjones) [inline package.json:20]: adds a the jest watch command

File: src/ReactPlayer/utils.test.js [hunk 2]
```diff
@@ -4,5 +4,5 @@ test('parseStartTime', () => {
   const url = 'https://youtube.com/video?t=1h14m30s'
   const timeInSeconds = parseStartTime(url)
 
-  expect(timeInSeconds).toEqual(219292)
+  expect(timeInSeconds).toEqual(4470)
```
Comment (theianjones) [inline src/ReactPlayer/utils.test.js:7]: 1 hour 14 minutes and 30 seconds = 4470 seconds not 219292 seconds 🤔

File: src/hooks/useInterval.test.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ test('useInterval', done => {
   const callback = called => {
     // because it's async, Jest throws errors so this traps it and makes it sensible
     try {
-      expect(called).toBe(false)
+      expect(called).toBe(true)
```
Comment (theianjones) [inline src/hooks/useInterval.test.js:14]: We pass true not false

File: src/hooks/useProgressInterval.test.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      playedSeconds: 10,
+    })
+
+    const playerRef = {
+      current: {
+        getDuration: () => 100,
+        getFractionPlayed: () => 0.1,
+        getFractionLoaded: () => 0.2,
+      },
+    }
```
Comment (theianjones) [inline src/hooks/useProgressInterval.test.js:28]: This is how I mocked a ref.

File: src/hooks/useProgressInterval.test.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import useProgressInterval from './useProgressInterval.js'
+
+const progressCallBackGenerator = (done, valuesToCheck) => progress => {
+  try {
+    expect(progress).toStrictEqual(valuesToCheck)
+    done()
+  } catch (error) {
+    done(error)
+  }
+}
```
Comment (theianjones) [inline src/hooks/useProgressInterval.test.js:12]: created this function so we dont have to write this try catch over and over

File: src/ReactPlayer/utils.test.js (hunk 2 above)
Comment (joelhooks) [reply src/ReactPlayer/utils.test.js:7]: all the tests I added were failed intentionally ;P

**Review by joelhooks — APPROVED** [review]

**Pull request closed without merging**